btree.go

```diff
@@ -321,11 +321,12 @@ func (b btreeIndex) Get(ng ngram) (ss simpleSection) {
 	off, sz := b.getBucket(bucketIndex)
 	bucket, err := b.file.Read(off, sz)
 	if err != nil {
-		return simpleSection{}
+		return simpleSection{}, stats
 	}
 
 	// find ngram in bucket
 	getNGram := func(i int) ngram {
+		stats.NgramsAccessed++
```

Comment:
@stefanhengl is this the magic place to capture this work? Am I missing other work done by btree?

Reply:
> Depends on what you are interested in. Here you track how many ngrams we have to decode during binary search. This should be less than 10 per ngram.

I was trying to find parts of the code which touch mmap regions and then count each access to that region. I assumed the binary search was the main thing which did that. But I realise now I may be missing all the work done by the btree to get to this "page".

> I think tracking the actual disk accesses would be interesting. The intention here is that the binary search happens all in memory.

How can we actually do that though? We just rely on mmap.

> I still haven't optimised the code for page boundaries, mainly because I didn't manage to reliably measure the page fault yet.

Yeah maybe Geoffrey's work has a graph we can look at. Either way I wanted to start looking at linux perf profiles to get a better idea. That mixed with getting a large corpus onto my linux desktop and I think I will be able to much more quickly experiment.